editor/src/components/navigator/navigator-item/navigator-item.tsx

```diff
@@ -513,21 +515,34 @@ export const NavigatorItem: React.FunctionComponent<
 
   const isGenerated = MetadataUtils.isElementGenerated(navigatorEntry.elementPath)
   const isDynamic = isGenerated || containsExpressions || isConditionalDynamicBranch
-  const isConditional = useEditorState(
+
+  const codeItemType: CodeItemType = useEditorState(
     Substores.metadata,
     (store) => {
-      if (isRegularNavigatorEntry(navigatorEntry)) {
-        return MetadataUtils.isElementPathConditionalFromMetadata(
-          store.editor.jsxMetadata,
-          navigatorEntry.elementPath,
-        )
-      } else {
-        return false
+      if (!isRegularNavigatorEntry(props.navigatorEntry)) {
+        return 'none'
+      }
+      const elementMetadata = MetadataUtils.findElementByElementPath(
+        store.editor.jsxMetadata,
+        props.navigatorEntry.elementPath,
+      )
+      if (MetadataUtils.isConditionalFromMetadata(elementMetadata)) {
+        return 'conditional'
+      }
+      if (MetadataUtils.isJSXMapExpressionFromMetadata(elementMetadata)) {
+        return 'map'
+      }
+      if (MetadataUtils.isExpressionOtherJavascriptFromMetadata(elementMetadata)) {
+        return 'code'
       }
+      return 'none'
     },
-    'NavigatorItem isConditional',
+    'NavigatorRowLabel codeItemType',
```

Comment:
I believe this should be `NavigatorItem`
```suggestion
    'NavigatorItem codeItemType',
```